From add8e0625d9248d9b6e998306541201dbf79805c Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Thu, 4 Apr 2024 15:00:47 +0200
Subject: [PATCH 50/82] arm64: dts: st: use power-domain property in GPU node
 in stm32mp255.dtsi

Set SCMI GPU power domain for GPU node in stm32mp255 device tree.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/linux.git db299e709b832f3bd7be054edd0097f45ab4f878 arm64: dts: st: use power-domain property in GPU node in stm32mp255.dtsi]
Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: I40a960a1e91e452190f6511c61c2d678ca0fa3f6
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/373467
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp255.dtsi | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp255.dtsi b/arch/arm64/boot/dts/st/stm32mp255.dtsi
index c444996b7d60d..fd699e5fedf34 100644
--- a/arch/arm64/boot/dts/st/stm32mp255.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp255.dtsi
@@ -3,6 +3,7 @@
  * Copyright (C) STMicroelectronics 2023 - All Rights Reserved
  * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
  */
+#include <dt-bindings/power/st,stm32mp25-power.h>
 #include "stm32mp253.dtsi"
 
 &ltdc {
@@ -62,7 +63,7 @@ gpu: gpu@48280000 {
 		resets = <&rcc GPU_R>;
 		clock-names = "axi", "core";
 		clocks = <&rcc CK_BUS_GPU>, <&rcc CK_KER_GPU>;
-		gpu-supply =  <&scmi_vddgpu>;
+		power-domains = <&scmi_devpd PD_SCMI_GPU>;
 		access-controllers = <&rifsc 79>;
 		status = "disabled";
 	};
-- 
2.51.0

